### ISSUE TITLE
Unable to debug in Visual Studio

### ISSUE THREAD

**Issue by alaatm** [dotnet/maui#1467]
I am not sure whether this bug report belongs here or not. I am trying this new framework and I have installed all required tools and everything works fine on the command line. However, when I try to debug in Visual Studio whether on local device or emulator, I always get an error message: **The path is not of a legal form**

The app does deploy on local device or emulator successfully but the debugger won't attach. Everything works as expected when debugging a xamarin forms app but not maui.

I am on the latest VS 2019 preview 16.11.0

**Comment by Ahmed0Ghaith** [comment]
you have to use Visual Studio 2019 16.11
https://visualstudio.microsoft.com/vs/preview/ to debug and deploy Maui

**Comment by alaatm** [comment]
Oops, sorry the version number was a typo. I am on the latest VS 2019 preview 16.11.0

**Comment by Ahmed0Ghaith** [comment]
ok use maui-check 
then use Preview 2 or greater
if it doesn't work 
create a Project through CMD
`dotnet new maui
`
and open it in IDE

**Comment by jitendravashishtha** [comment]
Same issue for me as well. Even if i try to create the application from command prompt or from visual studio.
[goldfish-address-space] allocate: Ask for block of size 0x100
[goldfish-address-space] allocate: ioctl allocate returned offset 0x397ffe000 size 0x2000
[HostConnection] HostComposition ext ANDROID_EMU_CHECKSUM_HELPER_v1 ANDROID_EMU_native_sync_v2 ANDROID_EMU_native_sync_v3 ANDROID_EMU_native_sync_v4 ANDROID_EMU_dma_v1 ANDROID_EMU_direct_mem ANDROID_EMU_host_composition_v1 ANDROID_EMU_host_composition_v2 ANDROID_EMU_vulkan ANDROID_EMU_deferred_vulkan_commands ANDROID_EMU_vulkan_null_optional_strings ANDROID_EMU_vulkan_create_resources_with_requirements ANDROID_EMU_YUV_Cache ANDROID_EMU_async_unmap_buffer ANDROID_EMU_vulkan_ignored_handles ANDROID_EMU_has_shared_slots_host_memory_allocator ANDROID_EMU_vulkan_free_memory_sync ANDROID_EMU_vulkan_shader_float16_int8 ANDROID_EMU_vulkan_async_queue_submit ANDROID_EMU_sync_buffer_data GL_OES_vertex_array_object GL_KHR_texture_compression_astc_ldr ANDROID_EMU_host_side_tracing ANDROID_EMU_async_frame_commands ANDROID_EMU_gles_max_version_2 
[monodroid] MonodroidRuntime::monodroid_dlopen ("/data/data/com.companyname.ShipmentTracking/files/.__override__/System.Runtime.Serialization.Formatters.dll.so", 0x1, 0xb687a760, 0x0)
[monodroid] MonodroidRuntime::monodroid_dlopen ("/__w/1/s/artifacts/obj/mono/Android.x86.Release/out/lib/mono/aot-cache/x86/System.Runtime.Serialization.Formatters.dll.so", 0x1, 0xb687a760, 0x0)
Loaded assembly: /data/data/com.companyname.ShipmentTracking/files/.__override__/System.Runtime.Serialization.Formatters.dll [External]
[hipmentTrackin] * Assertion at /__w/1/s/src/mono/mono/mini/debugger-engine.c:1088, condition `found_sp' not met
[libc] Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 32284 (.NET ThreadPool), pid 32242 (hipmentTracking)

**Comment by alaatm** [comment]
> ok use maui-check
> then use Preview 2 or greater
> if it doesn't work
> create a Project through CMD
> `dotnet new maui `
> and open it in IDE

To be clear, I have no issues with tooling, I have already went through the installation docs and I can create and run maui apps through the command line with no problems. My problem is with Visual Studio not being able to debug a maui app.

**Comment by alaatm** [comment]
@jitendravashishtha I think your issue is different than mine.

**Comment by Ahmed0Ghaith** [comment]
I think there issue with creating a project from  Visual Studio directly,
please create a project from the command and try to debug it on VS, in this case, the issue in Visual Studio.

**Comment by drasticactions** [comment]
> [hipmentTrackin] * Assertion at /__w/1/s/src/mono/mono/mini/debugger-engine.c:1088, condition `found_sp' not met

This is a bug in the .NET 6 Preview Runtime and the Android Interpreter (Which is used for C# Hot Reload). It's a known issue being addressed here, https://github.com/dotnet/runtime/pull/54763. Once that fix is merged in and released, it should work correctly again without workarounds.

If you want to work around it for now, disable the Interpreter in your csproj. It should be listed under `UseInterpreter`. This will disable C# Hot Reload, however. You can also disable XAML Hot Reload in the Visual Studio settings and that will also fix it. 

MAUI XAML Hot Reload isn't supported in 16.11+ builds anyway and will only work in Visual Studio 2022 (17+).

**Comment by jfversluis** [comment]
Since we are now on VS2022 I hope everything is fixed for you! Please update all the bits to the latest versions.
If it still doesn't seem to work, please open a new issue with all the details. Thanks!
